analyses/sex-prediction-from-RNASeq/sex-prediction-from-RNASeq.Rmd

```diff
@@ -0,0 +1,167 @@
+
+Use reported_gender column from histologies as the response.
+
+Extract reported_gender values from reported_gender column of histologies.  Put these value in a two-column dataframe
+  c("Kids_First_Biospecimen_ID", "reported_gender").
+
+```{r}
+
+reported_gender <- histologies[histologies$Kids_First_Biospecimen_ID %in% rownames(df), c("Kids_First_Biospecimen_ID", "reported_gender")]
+
+```
+
+Check sequence of rownames(df) and reported_gender[, 1].  reported_gender_response holds the reported_gender values in the same Kids_First_Biospecimen_ID sequence as rownames(df).
+
+```{r}
+
+match_index <- unlist(sapply(rownames(df), function(x) which(reported_gender[, 1] == x)))
+
+reported_gender_response <- reported_gender[match_index, 2]
+```
+
+Some values in reported_gender_response = "Not Available".  Eliminate from both reported_gender_response and df.
+
+```{r}
+reported_gender_response_hold <- reported_gender_response
+reported_gender_response <- reported_gender_response[reported_gender_response != "Not Available"]
+df <- df[reported_gender_response_hold != "Not Available", ]
+```
+
+
+Build predictive model for reported_gender_response.
+Hold out 30% of the samples to test accuracy after cross-validation tuning of 
+  lambda and alpha regularization parameters.
+
+```{r}
+
+train_percent <- 0.70
+set.seed(36354)
+
+train_set <- sample(1:nrow(df), floor(train_percent*nrow(df)))
+test_set <- setdiff(1:nrow(df), train_set)
+
+ptm <- proc.time()
+sex.cva <- cva.glmnet(df[train_set, ], reported_gender_response[train_set], standardize=TRUE, 
+                      alpha = seq(0, 1, len = 11)^3, family="binomial")
+proc.time() - ptm
+```
+
+Plot Deviance vs. log(lambda), one curve for each value of alpha.
+
+```{r}
+plot(sex.cva)
+```
+
+Plot CVloss against alpha.
+
+```{r}
+minlossplot(sex.cva, cv.type="min")
+```
+
+Although Hastie recommends against looking into the cva object, I could not retrieve optimal
+  values of lambda and alpha through code without doing so.
+
+the cva object contains a list, called modlist, of 11 objects, one for each of the alpha values tested.
+The CVloss plotted above is the minimum cvm value for each of the 11 objects in cva$modlist.
+The alpha corresponding to the minimum of these minimum cvm values is optimal.
+
+```{r}
+best_cvm_values <- sapply(sex.cva$modlist, function(x) min(x$cvm))
+best_alpha_index <- which(best_cvm_values == min(best_cvm_values))
+
+# look at sex.cva$alpha[best_alpha_index] to find the tuned value of alpha
+
+```
+
+The cva.modlist object corresponds to a given alpha value, and contains results for all the lambda values 
+  tested for the corresponding alpha value.
+
+The best performing lambda value is stored within the modlist object as lambda.min.
+
+We carry forward the predictors with non-zero coefficients in the optimal (minimum CVloss) model.
+
+```{r}
+best_fit <- sex.cva$modlist[[best_alpha_index]]
+
+#look at best_fit$lambda.min to find best lambda corresponding to tuned alpha
```

Comment:
I always thought the rule of thumb was to use `lambda.1se` rather than `lambda.min`: https://stats.stackexchange.com/questions/138569/why-is-lambda-within-one-standard-error-from-the-minimum-is-a-recommended-valu

Reply:
Ugh!  Thank you for catching this.  When we choose s = lambda.1se we get far fewer non-zero features (113 v.181) and the same predictive accuracy (95.8 v. 95.5).

lambda.1se used throughout is done and ready for add, commit.